code/modules/vr/vr_controller.dm

```diff
@@ -0,0 +1,66 @@
+//Control file with the backbone for the vr rooms and avatars
+
+var/datum/vr_rooms = list()
+var/roomcap = 20
+
+/datum/vr_room
+	var/name = null
+	var/datum/map_template/vr/template = null
+	var/datum/space_chunk/chunk = null
+	var/list/spawn_points = list()
+	var/expires = 1
+
+proc/make_vr_room(name, template, expires)
+	var/datum/vr_room/R = new
+	R.name = name
+	R.template = template
+	R.template = vr_templates[R.template]
+	R.chunk = space_manager.allocate_space(R.template.width, R.template.height)
+	R.expires = expires
+	vr_rooms[R.name] = R
+	R.template.load(locate(R.chunk.x,R.chunk.y,R.chunk.zpos), centered = FALSE)
+
+	for(var/atom/landmark in R.chunk.search_chunk(landmarks_list))
+		if(landmark.name == "avatarspawn")
+			R.spawn_points.Add(landmark)
+
+//datum/vr_room/process()
+
+
+//Control code for the avatars
+proc/build_virtual_avatar(mob/living/carbon/human/H, location, datum/map_template/vr/level/template)
+	if(!location)
+		return
+	var/mob/living/carbon/human/virtual_reality/vr_avatar
+	location = get_turf(location)
+	vr_avatar = new /mob/living/carbon/human/virtual_reality(location)
+	vr_avatar.set_species(H.species.name)
+	if(istype(H, /mob/living/carbon/human/virtual_reality))
+		var/mob/living/carbon/human/virtual_reality/V = H
+		vr_avatar.real_me = V.real_me
+	else
+		vr_avatar.real_me = H
+	vr_avatar.dna = H.dna.Clone()
```

Comment:
Syncing the DNA to the rest of the body and updating them as appropriate should make the hair update correctly as well. Look in `human_appearance.dm`.